webapp/channels/src/packages/mattermost-redux/src/reducers/entities/index.ts

```diff
@@ -62,3 +68,77 @@ export default combineReducers({
     sharedChannels,
     contentFlagging,
 });
+
+type EntitiesState = ReturnType<typeof entitiesReducers>;
+
+function handleFileRemovalFromPost(searchState: SearchState, action: MMReduxAction, postsState: PostsState) {
+    const updatedPost: Post = action.data;
+    const {posts} = postsState;
+
+    if (posts[updatedPost.id] && posts[updatedPost.id]?.file_ids) {
+        const oldPostFileIds = new Set(posts[updatedPost.id].file_ids);
+        const updatedPostFileIds = new Set(updatedPost.file_ids);
+        const getDeletedFiles = (oldPostFileIds: Set<string>, updatedPostFileIds: Set<string>) => {
+            const deletedFileIds = new Set();
+            for (const fileId of oldPostFileIds) {
+                if (!updatedPostFileIds.has(fileId)) {
+                    deletedFileIds.add(fileId);
+                }
+            }
+            return deletedFileIds;
+        }
+
+        // using this function instead of set.prototype.difference because the latest version of node (25)
+        // doesn't support it. Testing errors with message that difference is not a function
```

Comment:
This functionality could have been handled by `set.prototype.difference` but nodejs doesn't support that the tests failed when I used it

Reply:
According to https://compat-table.github.io/compat-table/es2016plus/, Node 22 and higher should actually support that, but we currently build MM with Node 20 so we're stuck with not being able to use the proper version in tests regardless. I'd consider polyfilling it in the test setup script, but I imagine that's going to break anyone writing test for plugins that use the Redux code, so we can leave this as-is until we upgrade Node further